core-web/apps/dotcms-ui/src/app/portlets/shared/dot-content-types-edit/components/dot-binary-settings/dot-binary-settings.component.spec.ts

```diff
@@ -0,0 +1,196 @@
+
+    const createComponent = createComponentFactory({
+        component: DotBinarySettingsComponent,
+        imports: [
+            NgFor,
+            FormsModule,
+            ReactiveFormsModule,
+            InputTextModule,
+            InputSwitchModule,
+            DividerModule,
+            DotMessagePipe
+        ],
+        providers: [
+            FormBuilder,
+            {
+                provide: DotFieldVariablesService,
+                useValue: {
+                    load: () =>
+                        of([
+                            {
+                                clazz: 'com.dotcms.contenttype.model.field.ImmutableStoryBlockField',
+                                fieldId: 'f965a51b-130a-435f-b646-41e07d685363',
+                                id: '9671d2c3-793b-41af-a485-e2c5fcba5fb',
+                                key: 'systemOptions',
+                                value: systemOptions
+                            },
+                            {
+                                clazz: 'com.dotcms.contenttype.model.field.ImmutableStoryBlockField',
+                                fieldId: 'f965a51b-130a-435f-b646-41e07d685363',
+                                id: '9671d2c3-793b-41af-a485-e2c5fcba5fb',
+                                key: 'accept',
+                                value: 'image/*'
+                            }
+                        ]),
+                    save: () => of([]),
+                    delete: () => of([])
+                }
+            },
+            {
+                provide: DotMessageService,
+                useValue: messageServiceMock
+            },
+            {
+                provide: DotHttpErrorManagerService,
+                useValue: {
+                    handle: () => of([])
+                }
+            }
+        ]
+    });
+
+    beforeEach(async () => {
+        spectator = createComponent({});
+        dotFieldVariableService = spectator.inject(DotFieldVariablesService);
+        dotHttpErrorManagerService = spectator.inject(DotHttpErrorManagerService);
+
+        component = spectator.component;
+    });
+
+    it('should setup form values', async () => {
+        expect(component.form.get('accept').value).toBe('image/*');
+        expect(component.form.get('allowURLImport').value).toBe(false);
+        expect(component.form.get('allowCodeWrite').value).toBe(true);
+        expect(component.form.get('allowFileNameEdit').value).toBe(false);
+    });
+
+    it('should emit changeControls when isVisible input is true', () => {
+        spyOn(component.changeControls, 'emit');
+
+        spectator.setInput('isVisible', true);
+
+        expect(component.changeControls.emit).toHaveBeenCalled();
+    });
+
+    it('should emit valid output on form change', () => {
+        spyOn(component.valid, 'emit');
+
+        component.form.get('accept').setValue('text/*');
+
+        expect(component.valid.emit).toHaveBeenCalled();
+    });
```

Comment:
Should we use the DOM element to trigger the `valueChanges` of the `form` instead of changing the value of the form manually?